### PR TITLE
Build system simplification

### PR DESCRIPTION
**Pull request by mkuthan**

### CODE REVIEW

File: src/client/swiv-entry.scss [hunk 1]
```diff
@@ -14,10 +14,10 @@
  * limitations under the License.
  */
 
-@import '../../node_modules/laborer/common/style/reset';
-@import '../../node_modules/laborer/common/style/sanity';
-@import '../../node_modules/laborer/common/style/open-sans-regular';
-@import '../../node_modules/laborer/common/style/open-sans-600';
+@import '../../lib/laborer/common/style/reset';
```
Comment (mkuthan) [inline src/client/swiv-entry.scss]: Do we need laborer anymore?

File: src/client/utils/ajax/ajax.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ export class Ajax {
             throw new Error(xhr.responseText || 'connection fail');
           }
         }
-      });
+      }) as any;
```
Comment (pguzik) [inline src/client/utils/ajax/ajax.ts]: do we need such cast ?

File: README.md [hunk 3]
```diff
@@ -31,33 +31,33 @@ this is how Turnilo emerged.
...
+npm run start -- --examples
 ```
 
 ### Run with Druid
 
 Next connect Turnilo to Druid cluster by simply pointing it to broker host:
 
 ```
-./bin/swiv --druid your.druid.broker.host:8082
+npm run start -- --druid your.druid.broker.host:8082
```
Comment (pguzik) [inline README.md]: i would write your.druid.broker.host (default to 8082)

File: package.json [hunk 4]
```diff
@@ -14,72 +14,99 @@
...
+    "start-prod": "NODE_ENV=production ./bin/swiv",
+    "build": "npm run clean | tsc | webpack --config config/webpack.dev.js",
+    "build:prod": "npm run clean | tsc | webpack --config config/webpack.prod.js",
+    "clean": "rm -rf ./build/*",
+    "tsc": "tsc -t ES2015 --lib \"ES2015\",\"DOM\",\"ScriptHost\""
   },
   "dependencies": {
-    "body-parser": "1.15.2",
-    "browser-filesaver": "1.1.1",
+    "body-parser": "^1.18.2",
```
[the remark this answers is not in the document]
Comment (pguzik) [reply package.json]: +1 let the libs stay in newest working version

File: src/client/components/router/router.mocha.tsx [hunk 5]
```diff
@@ -202,7 +202,7 @@ describe('Router', () => {
     });
 
 
-    it('initializes to the location', (done) => {
+    it('initializes to the location', (done:any) => {
```
Comment (pguzik) [inline src/client/components/router/router.mocha.tsx]: do we need any type here ?

File: src/client/utils/test-utils/find-dom-node.ts [hunk 6]
```diff
@@ -15,10 +15,13 @@
...
 
 import * as ReactDOM from 'react-dom';
-import * as TestUtils from 'react-addons-test-utils';
+import * as TestUtils from 'react-dom/test-utils';
 import { BodyPortal } from '../../components/body-portal/body-portal';
 
-export function findDOMNode(element: __React.Component<any, any>): Element {
-  var portal = TestUtils.scryRenderedComponentsWithType(element, BodyPortal)[0];
+export function findDOMNode(element: __React.Component<any, any> | Element ): Element {
+/*  if(element instanceof Element) {
```
Comment (pguzik) [inline src/client/utils/test-utils/find-dom-node.ts]: why this is commented out ?

File: src/client/views/cube-view/cube-view.mocha.tsx [hunk 7]
```diff
@@ -70,7 +70,7 @@ describe('CubeView', () => {
     var stub = sinon.stub(localStorage, "get");
     stub.withArgs('is-multi-measure').returns(undefined);
 
-    var initialCubeView = TestUtils.renderIntoDocument(
+    var initialCubeView:any = TestUtils.renderIntoDocument(
```
Comment (pguzik) [inline src/client/views/cube-view/cube-view.mocha.tsx]: is any type a good thing here ?

File: src/client/visualizations/index.tsx [hunk 8]
```diff
@@ -24,7 +24,8 @@ import { LineChart } from './line-chart/line-chart';
 import { BarChart } from './bar-chart/bar-chart';
 import { Geo } from './geo/geo';
 
-const VIS_COMPONENTS: Array<typeof BaseVisualization> = [
+// TODO, back to: const VIS_COMPONENTS: Array<typeof BaseVisualization> = [
```
Comment (pguzik) [inline src/client/visualizations/index.tsx]: why is this here TODO ?

File: .travis.yml [hunk 9]
```diff
@@ -1,9 +1,6 @@
 language: node_js
 node_js:
-  - "4.0"
   - "6.1"
```
Comment (mkuthan) [inline .travis.yml]: can we update to the latest stable version, e.g 8.x?

File: package.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "yahoo-swiv",
+  "name": "turnilo",
   "version": "0.9.42",
```
Comment (mkuthan) [inline package.json]: let's start with version 1.0.0

File: package.json [hunk 11]
```diff
@@ -87,19 +92,24 @@
     "@types/supertest": "2.0.4",
     "@types/webpack-env": "1.13.3",
     "awesome-typescript-loader": "3.4.1",
+    "chai": "4.1.2",
     "create-react-class": "15.6.2",
     "css-loader": "0.28.7",
+    "css-modules-require-hook": "^4.2.3",
```
Comment (mkuthan) [inline package.json]: why ^...

File: package.json [hunk 12]
```diff
@@ -87,19 +92,24 @@
...
     "jsdom": "9.4.2",
     "mocha": "4.1.0",
     "node-sass": "4.7.2",
     "react-addons-test-utils": "15.6.2",
     "react-hot-loader": "3.1.3",
+    "react-test-renderer": "16.2.0",
     "regex-replace-loader": "1.0.0",
-    "rewire": "3.0.2",
+    "rewire": "2.5.2",
+    "rimraf": "^2.6.2",
```
Comment (mkuthan) [inline package.json]: why ^...

File: src/client/components/date-range-picker/date-range-picker.mocha.tsx [hunk 13]
```diff
@@ -47,6 +47,7 @@ describe('DateRangePicker', () => {
     expect((ReactDOM.findDOMNode(renderedComponent) as any).className, 'should contain class').to.contain('date-range-picker');
   });
 
+/*
```
Comment (mkuthan) [inline src/client/components/date-range-picker/date-range-picker.mocha.tsx]: why this is commented out?

File: src/client/components/date-range-picker/date-range-picker.mocha.tsx [hunk 14]
```diff
@@ -60,7 +61,9 @@ describe('DateRangePicker', () => {
         />);
     }).to.throw('start time must be round');
   });
+*/
 
+/*
```
Comment (mkuthan) [inline src/client/components/date-range-picker/date-range-picker.mocha.tsx]: why this is commented out?
